examples/mugen/retrieval/video_clip.py

```diff
@@ -72,14 +81,115 @@ class VideoEncoder(nn.Module):
+        super().__init__()
+        self.linear1 = nn.Linear(in_dim, out_dim, bias=False)
+        self.gelu = nn.GELU()
+        self.linear2 = nn.Linear(out_dim, out_dim, bias=False)
+        self.drop = nn.Dropout(dropout_prob)
+        self.layer_norm = nn.LayerNorm(out_dim)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        embed1 = self.linear1(x)
+        embed2 = self.gelu(embed1)
+        embed2 = self.linear2(embed2)
+        embed2 = self.drop(embed2)
+        embeds = self.layer_norm(embed1 + embed2)
+        return embeds
+
+
+def videoclip(
+    text_pretrained: bool = True,
+    text_trainable: bool = True,
+    text_model_name: str = "distilbert-base-uncased",
+    text_model_config: Optional[Dict[str, Any]] = None,
+    text_padding_value: int = 0,
+    video_trainable: bool = True,
+    proj_out_dim: int = 256,
+    proj_dropout: float = 0.1,
+) -> CLIPArchitecture:
+    """Create MUGEN's video-text CLIP model with a S3D-backed video encoder and DistilBERT-backed text encoder.
+        MUGEN paper: https://arxiv.org/abs/2204.08058
+
+    Args:
+        text_pretrained (bool): whether to use a pretrained text encoder or not.
+            Defaults to True.
+        text_trainable (bool): whether the text encoder's weights should be trainable.
+            Defaults to True.
+        text_model_name (str): name of pretrained model, used when pretrained is True.
+            Defaults to "distilbert-base-uncased", Hugging Face's standard DistilBERT model.
+        text_model_config (Optional[Dict[str, Any]]): model config for DistilBERT, used when pretrained is False
+            Defaults to None, indicating the default DistilBERT config.
+        text_padding_value (int): value that was used to pad the input text.
+            Defaults to 0, Hugging Face's BERT pad token.
+        video_trainable (bool): whether the video encoder's weights should be trainable.
+            Defaults to True.
+        proj_out_dim (int): output dimension to project both encoders' outputs to.
+            Defaults to 256, the value used by MUGEN.
+        proj_dropout (float): dropout probability in the projection layers.
+            Defaults to 0.1, the value used by MUGEN.
+
+    Returns:
+        CLIPArchitecture: CLIP model using MUGEN's video encoder and text encoder.
+
+    """
+    text_model = TextEncoder(
+        pretrained=text_pretrained,
+        trainable=text_trainable,
+        model_name=text_model_name,
+        model_config=text_model_config,
+        padding_value=text_padding_value,
+    )
+    text_encoder = nn.Sequential(
+        text_model,
+        Projection(
+            text_model.model.config.dim, out_dim=proj_out_dim, dropout_prob=proj_dropout
+        ),
+    )
+    video_encoder = nn.Sequential(
+        VideoEncoder(
+            trainable=video_trainable,
+        ),
+        Projection(1024, out_dim=proj_out_dim, dropout_prob=proj_dropout),
```

Comment:
Hard coding intermediate dimensions can be error prone as it assumes what's returned by the previous layer. 
Is `1024` the last dim of `VideoEncoder` (the concat dim of [`Mixed5c`](https://github.com/pytorch/vision/blob/6512146e447b69cc9fb379eb05e447a17d7f6d1c/test/test_datasets_utils.py#L20-L29))? 
What if we create an attribute `VideoEncoder.out_dim` or `S3D.out_dim` so if there's any inconsistency the error will be self-contained within the module that causes the error.

Reply:
yup, 1024 is the size of the output of Mixed5c. I'll add the out_dim attributes to both encoders